src/Distribution/Hackage/DB/Parsed.hs

```diff
@@ -30,7 +31,7 @@ type HackageDB = Map PackageName PackageData
 
 type PackageData = Map Version VersionData
 
-data VersionData = VersionData { cabalFile :: !GenericPackageDescription
+data VersionData = VersionData { cabalFileRevisions :: NonEmpty GenericPackageDescription
```

Comment:
Storing different revisions in a *list* means that I cannot easily find a given revision. Before this change, you had essentially a lookup `PackageName -> Version -> GenericPackageDescription`, which would always give you the latest available revision. If we'd like to make the code aware of revisions, then IMHO it should come in the form of a lookup:

    PackageName -> Version -> Revision -> GenericPackageDescription

One possible alternative interface could be to extend our notion of `Version` with something that is aware of revisions. I suppose that would end up being more space efficient that to create a whole lot of singletons `Map`s for the majority of package versions, which have never been revised.